packages/perseus/src/widgets/radio/radio-component.new.tsx

```diff
@@ -0,0 +1,476 @@
+                crossedOut: false,
+                readOnly: false,
+                highlighted: false,
+                rationaleShown: false,
+                correctnessShown: false,
+                previouslyAnswered: false,
+            }));
+        }
+
+        const choicesProp: ReadonlyArray<ChoiceType> = choices.map(
+            (choice, i) => {
+                const content =
+                    choice.isNoneOfTheAbove && !choice.content
+                        ? strings.noneOfTheAbove
+                        : choice.content;
+
+                const {
+                    selected,
+                    crossedOut,
+                    rationaleShown,
+                    correctnessShown,
+                    readOnly,
+                    highlighted,
+                    previouslyAnswered,
+                } = choiceStates[i];
+
+                const reviewChoice = this.props.reviewModeRubric?.choices[i];
+
+                return {
+                    content: this._renderRenderer(content),
+                    checked: selected,
+                    // Current versions of the radio widget always pass in the
+                    // "correct" value through the choices. Old serialized state
+                    // for radio widgets doesn't have this though, so we have to
+                    // pull the correctness out of the review mode scoring data. This
+                    // only works because all of the places we use
+                    // `restoreSerializedState()` also turn on reviewMode, but is
+                    // fine for now.
+                    // TODO(emily): Come up with a more comprehensive way to solve
+                    // this sort of "serialized state breaks when internal
+                    // structure changes" problem.
+                    correct:
+                        choice.correct === undefined
+                            ? !!reviewChoice && !!reviewChoice.correct
+                            : choice.correct,
+                    disabled: readOnly,
+                    hasRationale: !!choice.clue,
+                    rationale: this._renderRenderer(choice.clue),
+                    showRationale: rationaleShown,
+                    showCorrectness: correctnessShown,
+                    isNoneOfTheAbove: !!choice.isNoneOfTheAbove,
+                    revealNoneOfTheAbove: !!(
+                        this.props.questionCompleted && selected
+                    ),
+                    crossedOut,
+                    highlighted,
+                    previouslyAnswered: previouslyAnswered,
+                };
+            },
+        );
+
+        // This is a temporary console log to ensure that the new radio component
+        // is being rendered. It will be removed after the feature flag is no longer needed.
+        // TODO(LEMS-2994): Remove this console log.
+        // eslint-disable-next-line no-console
+        console.log("This is the new radio component.");
```

Comment:
I thought this would be helpful for us to know when we're seeing the new component, at least until we start implementing the visual upgrades.